File: src/dotnet/PloyBus.Tests/Transport/InMemory/TestEndpoint.cs
using System.Diagnostics;
using PolyBus.Transport.Transactions;

namespace PolyBus.Transport.InMemory;

[DebuggerStepThrough]
class TestEndpoint
{
    public Func<IncomingTransaction, Task> OnMessageReceived { get; set; } = _ => Task.CompletedTask;
    public IPolyBus Bus { get; set; } = null!;
    public PolyBusBuilder Builder { get; } = new();
    public InMemoryEndpoint Transport => (InMemoryEndpoint)Bus.Transport;
    public async Task Handler(IncomingTransaction transaction, Func<Task> next)
    {
        await OnMessageReceived(transaction);
        await next();
    }
}

File: src/kotlin/src/test/kotlin/polybus/transport/inmemory/TestEndpoint.kt
package polybus.transport.inmemory

import polybus.IPolyBus
import polybus.PolyBusBuilder
import polybus.transport.transactions.IncomingTransaction

class TestEndpoint {
    var onMessageReceived: suspend (IncomingTransaction) -> Unit = {}
    lateinit var bus: IPolyBus
    val builder: PolyBusBuilder = PolyBusBuilder()
    val transport: InMemoryEndpoint
        get() = bus.transport as InMemoryEndpoint

    suspend fun handler(transaction: IncomingTransaction, next: suspend () -> Unit) {
        onMessageReceived(transaction)
        next()
    }
}
